app/services/widgets.ts

```diff
@@ -3,30 +3,45 @@ import { Service } from './service';
 import { Inject } from '../util/injector';
 import { UserService, requiresLogin } from './user';
 import { TPlatform } from './platforms';
-import { ScenesService, SceneItem } from './scenes';
+import { ScenesService, SceneItem, Scene } from './scenes';
 import { SourcesService } from './sources';
 import { VideoService } from './video';
 import { HostsService } from './hosts';
 import { ScalableRectangle, AnchorPoint } from '../util/ScalableRectangle';
 import namingHelpers from '../util/NamingHelpers';
+import { Dictionary } from 'lodash';
+import fs from 'fs';
 
 export enum WidgetType {
-  AlertBox,
-  DonationGoal,
-  FollowerGoal,
-  SubscriberGoal,
-  BitGoal,
-  DonationTicker,
-  ChatBox,
-  EventList,
-  TheJar,
-  ViewerCount,
-  StreamBoss,
-  Credits
+  AlertBox = 0,
+  DonationGoal = 1,
+  FollowerGoal = 2,
+  SubscriberGoal = 3,
+  BitGoal = 4,
+  DonationTicker = 5,
+  ChatBox = 6,
+  EventList = 7,
+  TheJar = 8,
+  ViewerCount = 9,
+  StreamBoss = 10,
+  Credits = 11
```

Comment:
This `enum` is persisted in a number of locations now.  I have switched to hard coding the values to make it obvious that the values of existing should not be changed, otherwise it will cause lots of issues.

In the future, I would like to use string enums for stuff like this, but we will need to ugprade Typescript.

Reply:
We upgraded TypeScript some months ago to 2.4. So we can use string enums now. Last time I told only about upgrade to 2.6

Reply:
oh ok cool